### ISSUE TITLE
test with clangd 

### ISSUE THREAD

**Issue by teto** [tjdevries/nvim-langserver-shim#17]
Very eager to see a server for C/C++ so I installed clangd from source even if it's not robust.

Here is some output I get when running on nvim source code. I hope we can have a cleaner error.
```
Erreur détectée en traitant function <SNR>185_on_stdout[4]..<SNR>184__on_lsp_stdout :                                                                              
ligne   56 :                                                                                                                                                       
E474: Unidentified byte: jsonrpc":"2.0","id":3,"error":{"code":-32601}}Co
```
```
debug: Sending Request: 15, {'id': 1, 'jsonrpc': '2.0', 'method': 'initialize', 'params': {'capabilities': {}, 'rootPath': 'file:///home/teto/neovim'}}
debug: ( 15:         stderr): ['<-- {"id": 1, "jsonrpc": "2.0", "method": "initialize", "params": {"capabilities": {}, "rootPath": "file:///home/teto/neovim"}}', '']
debug: ( 15:         stderr): ['--> {"jsonrpc":"2.0","id":1,"result":{"capabilities":{', '          "textDocumentSync": 1,', '          "documentFormattingProvider": true,', '          "documentRangeFormattingProvider": true,', '          "documentOnTypeFormattingProvider": {"firstTriggerCharacter":"}","moreTriggerCharacter":[]},', '          "codeActionProvider": true,', '          "completionProvider": {"resolveProvider": false, "triggerCharacters": [".",">"]}', '        }}}', '']
debug: Sending Request: 15, {'id': 2, 'jsonrpc': '2.0', 'method': 'textDocument/definition', 'params': {'textDocument': {'uri': 'file:///home/teto/neovim/src/nvim/menu.c'}, 'position': {'character': 8, 'line': 776}}}
debug: ( 15:         stderr): ['<-- {"id": 2, "jsonrpc": "2.0", "method": "textDocument/definition", "params": {"textDocument": {"uri": "file:///home/teto/neovim/src/nvim/menu.c"}, "position": {"character": 8, "line": 776}}}', 'Method ignored.', '--> {"jsonrpc":"2.0","id":2,"error":{"code":-32601}}', '']
debug: Sending Request: 15, {'id': 3, 'jsonrpc': '2.0', 'method': 'textDocument/definition', 'params': {'textDocument': {'uri': 'file:///home/teto/neovim/src/nvim/menu.c'}, 'position': {'character': 9, 'line': 770}}}
debug: ( 15:         stderr): ['<-- {"id": 3, "jsonrpc": "2.0", "method": "textDocument/definition", "params": {"textDocument": {"uri": "file:///home/teto/neovim/src/nvim/menu.c"}, "position": {"character": 9, "line": 770}}}', 'Method ignored.', '--> {"jsonrpc":"2.0","id":3,"error":{"code":-32601}}', '']
```

Thanks for the plugin. Looking forward to use it once clangd gets released.

**Comment by tjdevries** [comment]
hi @teto! Thanks for trying out the plugin. I'm actually working on rewriting this in lua here: https://github.com/tjdevries/neovim/tree/lsp

I'm going to make a PR soon, now that Lua has been merged into neovim. :) After I'm done with that, I'm going to deprecate this plugin, since there will be a better, faster and supported one :)

**Comment by teto** [comment]
awesome !

2017-05-09 2:07 GMT+02:00 TJ DeVries <notifications@github.com>:

**Comment by tjdevries** [comment]
PR is here: https://github.com/neovim/neovim/pull/6856

Still kind of far from doing everything this plugin could do, but I'm excited about the progress.

**Comment by nickdiego** [comment]
Hi @tjdevries !! Thanks for you effort on getting LSP support in neovim!!

BTW, any news about it?

cheers,
Nick

**Comment by tjdevries** [comment]
Hi @nickdiego 

I'm currently waiting on a few other PRs in Neovim before I resume a lot of work on that PR. Most of them revolve around using jobs in lua. This will make the implementation of the server much cleaner.
I don't expect it to get merged (or finished for that matter :) ) very soon.

is there anything in particular you were wondering about?

TJ

**Comment by nickdiego** [comment]
Cool @tjdevries !

>is there anything in particular you were wondering about?

Not at all, just excited with this promising LSP support in neovim. Just tried for the 2nd to use [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim) with no success (tried with Eclipse`s Java Lang Server) :(
For now, do you recommend any LSP plugin that works well enough with Java/Android and C++ projects? Not tried to compile `clangd` yet.

Anyway, looking forward to see LSP support in neovim!
Nice job.

**Comment by prabirshrestha** [comment]
In the mean time could you try https://github.com/prabirshrestha/vim-lsp. The wiki on the side bar has some instructions on how to setup. If you get one working feel free to update the wiki. https://github.com/prabirshrestha/vim-lsp/wiki/Servers

**Comment by nickdiego** [comment]
>In the mean time could you try https://github.com/prabirshrestha/vim-lsp. The wiki on the side bar has some instructions on how to setup. If you get one working feel free to update the wiki. https://github.com/prabirshrestha/vim-lsp/wiki/Servers

Thanks @prabirshrestha , I'll give it a try for sure :)

**Comment by teto** [comment]
cleaning up my issues.
